File: src/melody_sync/core/scanner/scanner.py
from pathlib import Path

from melody_sync.core.scanner.discovery import discover
from melody_sync.core.scanner.metadata import read_metadata
from melody_sync.models.song import Song


def scan(directory: Path) -> list[Song]:
    """
    Percorre uma biblioteca musical e retorna uma lista de objetos Song.
    """

    songs: list[Song] = []

    for path in discover(directory):
        song = Song(
            path=path,
            size=path.stat().st_size,
        )

        songs.append(read_metadata(song))

    return songs

File: melody-sync-core/src/main/kotlin/com/melodysync/scanner/Scanner.kt
package com.melodysync.scanner

import com.melodysync.model.Song
import java.nio.file.Files
import java.nio.file.Path

fun scan(directory: Path): List<Song> =
    discover(directory).map { path ->
        val size = Files.size(path)
        readMetadata(Song(path = path, size = size))
    }
